samples/highcharts/website/xmas-2021/demo.js

```diff
@@ -0,0 +1,3937 @@
+    const getStarData = function () {
+        for (let ii = 0; ii <= 40; ++ii) {
+            const newX = randomIntFromInterval(0, 10.5);
+            const newY = randomIntFromInterval(5, 8);
+            starData.push([newX, newY]);
+        }
+
+    };
+    getStarData();
+
+    Highcharts.getJSON(
+        'https://cdn.jsdelivr.net/gh/highcharts/highcharts@2e11000c966a20f08afc4e0927b91df99821de99/samples/data/world-countries.topo.json',
+        topology => {
+
+            // Convert the topoJSON feature into geoJSON
+            const geojson = window.topojson.feature(
+                topology,
+                // For this demo, get the first of the named objects
+                topology.objects[Object.keys(topology.objects)[0]]
+            );
+            geojson.copyrightUrl = topology.copyrightUrl;
+            geojson.copyrightShort = topology.copyrightShort;
+
+            chart = Highcharts.mapChart('ball-chart', {
+                chart: {
+                    map: geojson,
+                    margin: 0,
+                    backgroundColor: 'url(#bg-gradient)',
+                    events: {
+                        load: function () {
+                            const chart = this;
+                            let count = 0;
+
+                            chart.series[3].points[0].setState('hover');
+
+                            if (!big) {
+                                chart.update({
+                                    plotOptions: {
+                                        pie: {
+                                            center: [70, 5]
+
+                                        }
+                                    }
+                                });
+                            }
+
+
+                            //spin
+                            setTimeout(function () {
+                                setInterval(function () {
+                                    if (count < 360) {
+                                        chart.update({
+                                            mapView: {
+                                                projection: {
+                                                    name: 'Orthographic',
+                                                    rotation: [count, -30]
+                                                }
+                                            }
+                                        });
+                                        count = count + 1;
+                                    } else {
+                                        chart.update({
+                                            mapView: {
+                                                projection: {
+                                                    name: 'Orthographic',
+                                                    rotation: [count, -30]
+                                                }
+                                            }
+                                        });
```

Comment:
Not a big deal, but this block can be combined with the previous by just using the modulo operator for count: `rotation: [count % 360, -30]`. Which means that the count will continue to increase past 360, but the rotation will always be between 0 and 360.